commit 9981c558f6c7592b1ee652915e3baf09c33f7f27
Author: Richard Hughes <richard@hughsie.com>
Date:   Thu Jul 8 09:43:04 2010 +0100

    yum: wrap _getEVR in a try,catch block to deal with invalid version numbers. Fixes rh#612360

diff --git a/backends/yum/yumBackend.py b/backends/yum/yumBackend.py
index f7bd2e4..90c4062 100755
--- a/backends/yum/yumBackend.py
+++ b/backends/yum/yumBackend.py
@@ -2846,7 +2846,11 @@ class PackageKitYumBackend(PackageKitBaseBackend, PackagekitPackage):
 
                     # is older than what we have already?
                     if instpkg:
-                        evr = _getEVR(version[1])
+                        evr = ('0', '0', '0')
+                        try:
+                            evr = _getEVR(version[1])
+                        except Exception, e:
+                            pass
                         if evr == ('0', '0', '0'):
                             changelog += ";*Could not parse header:* '%s', *expected*: 'Firstname Lastname <email@account.com> - version-release';" % header
                         rc = rpmUtils.miscutils.compareEVR((instpkg.epoch, instpkg.version, instpkg.release.split('.')[0]), evr)
